src/Main.cpp

```diff
@@ -1,10 +1,466 @@
+    message["streamIds"] = Json::arrayValue;
+    message["target"] = "leave_session";
+    message["type"] = 1;
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void pong(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder,
+    string const& local_peer_id
+)
+{
+    Json::Value message;
+    message["protocol"] = "one-to-one";
+    message["to"] = decoder.getFrom();
+    message["action"] = "pong";
+    message["data"]["from"] = local_peer_id;
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void offerAnswerRustyMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["target"] = decoder.getTo();
+    message["caller"] = decoder.getFrom();
+    message["sdp_message"]["type"] = decoder.getActionType();
+    message["sdp_message"]["sdp"] = decoder.getDescription();
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void candidateRustyMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    RustySignalMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["target"] = decoder.getTo();
+    message["candidate"] = decoder.getCandidate();
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void offerAnswerSignalMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["protocol"] = "one-to-one";
+    message["to"] = decoder.getTo();
+    message["action"] = decoder.getActionType();
+    message["data"]["from"] = decoder.getFrom();
+    message["data"]["description"] = decoder.getDescription();
+
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+void candidateSignalMessageParser(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder
+)
+{
+    Json::Value message;
+    message["protocol"] = "one-to-one";
+    message["to"] = decoder.getTo();
+    message["action"] = decoder.getActionType();
+    message["data"]["from"] = decoder.getFrom();
+    message["data"]["candidate"] = decoder.getDescription();
+    if (auto ws = make_weak_ptr(websocket).lock())
+    {
+        Json::FastWriter fast;
+        ws->send(fast.write(message));
+    }
+}
+
+bool initialHandShakeFinalized(
+    shared_ptr<rtc::WebSocket>& websocket,
+    SignalRMessageDecoder& decoder,
+    string& local_peer_id
+)
+{
+    if (decoder.isEmpty())
+    {
+        sendSessionCheck(websocket, local_peer_id);
+        return false;
+    }
+    else if (decoder.checkSessionList())
```

Comment:
I would imagine you would get the client id from the session list, maybe it isnt really needed to perform the session check?

Reply:
The check is basically to confirm that the deep_trekker got our `local_peer_id` as `client_id`. So we should get the id from the list and compare with the local_peer_id.